gc/base/standard/Scavenger.cpp

```diff
@@ -2298,21 +2298,25 @@ MM_Scavenger::pruneRememberedSetOverflow(MM_EnvironmentStandard *env)
 				if(_extensions->objectModel.isRemembered(objectPtr)) {
 					bool shouldBeRemembered = false;
 
-					/* Re-scan the tenured for objects that should be remembered.
-					 * No copying will be done. */
-					shouldBeRemembered = shouldRememberObject(env, objectPtr);
-
-					/* The remembered state of a class object also depends on the class statics */
-					if (_extensions->objectModel.hasIndirectObjectReferents(env->getLanguageVMThread(), objectPtr)) {
-						shouldBeRemembered |= _cli->scavenger_hasIndirectReferentsInNewSpace(env, objectPtr);
-					}
 #if !defined(OMR_GC_CONCURRENT_SCAVENGER)
 					/* unconditionally remember any recently referenced objects */
 					if (processRememberedThreadReference(env, objectPtr)) {
 						Trc_MM_ParallelScavenger_scavengeRememberedSet_keepingRememberedObject(env->getLanguageVMThread(), objectPtr, _extensions->objectModel.getRememberedBits(objectPtr));
 						shouldBeRemembered = true;
 					}
-#endif
+#endif /* !defined(OMR_GC_CONCURRENT_SCAVENGER) */
+
+					if (!shouldBeRemembered) {
+						/* The remembered state of a class object also depends on the class statics */
```

Comment:
Comment was modified in the revision that I checked out. Can restore this (and move tracepoint as described above if you want). Let me know one way or other (do nothing, fix comment only, or fix comment(s) and move tracepoint. Also pls provide direction regarding removal of setRememberedSetOverflow() out of SublistFragment and into MM_Scavenger.